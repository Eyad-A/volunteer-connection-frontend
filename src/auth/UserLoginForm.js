import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Alert from "../common/Alert";

/** User login form
 * Shows a login form 
 * Routed at /login 
 */

function UserLoginForm({ loginUser }) {

  const [formData, setFormData] = useState({
    username: "",
    password: "",
  });
  const [formErrors, setFormErrors] = useState([]);
  const history = useHistory();

  // Handle form submission 
  async function handleSubmit(evt) {
    evt.preventDefault();
    let results = await loginUser(formData);
    if (results.success) {
      history.push("/companies");
    } else {
      setFormErrors(results.errors);
    }
  }

  // Handle change function 
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(d => ({ ...d, [name]: value }));
  }

  return (
    <div className="container my-5">
      <div className="row">
        <div className="col-lg-2">

        </div>
        <div className="col-lg-10">
          <h1>User Login Form</h1>
          <div className="my-3">
            <form onSubmit={handleSubmit}>
              <div className="col-lg-9 my-2">
                <input
                  name="username"
                  className="form-control"
                  placeholder="Username"
                  value={formData.username}
                  onChange={handleChange}
                  required
                />
              </div>
              <div className="col-lg-9 my-2">
                <input
                  name="password"
                  className="form-control"
                  placeholder="Password"
                  type="password"
                  value={formData.password}
                  onChange={handleChange}
                  required
                />
              </div>

              {formErrors.length
                ? <Alert type="danger" message={formErrors} />
                : null
              }

              <button className="btn btn-lg btn-primary my-3">
                Submit
              </button>
            </form>
          </div>
        </div>
        <div className="col-lg-2">

        </div>
      </div>
    </div>
  );
}

export default UserLoginForm;