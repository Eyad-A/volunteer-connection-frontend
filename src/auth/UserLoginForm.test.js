import React from "react";
import { render } from "@testing-library/react";
import UserLoginForm from "./UserLoginForm";
import { MemoryRouter } from "react-router";

it("renders without crashing", function() {
    render(
        <MemoryRouter>
            <UserLoginForm />
        </MemoryRouter>
    );
});

it("matches snapshot", function() {
    const { asFragment } = render(
        <MemoryRouter>
            <UserLoginForm />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});