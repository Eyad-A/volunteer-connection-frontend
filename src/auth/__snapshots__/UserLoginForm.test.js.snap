// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container my-5"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-2"
      />
      <div
        class="col-lg-10"
      >
        <h1>
          User Login Form
        </h1>
        <div
          class="my-3"
        >
          <form>
            <div
              class="col-lg-9 my-2"
            >
              <input
                class="form-control"
                name="username"
                placeholder="Username"
                required=""
                value=""
              />
            </div>
            <div
              class="col-lg-9 my-2"
            >
              <input
                class="form-control"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-lg btn-primary my-3"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
      <div
        class="col-lg-2"
      />
    </div>
  </div>
</DocumentFragment>
`;
